gengokit/middlewares/templates/endpoints.go

```diff
@@ -4,21 +4,21 @@ const EndpointsBase = `
 package middlewares
 
 import (
-	"github.com/go-kit/kit/endpoint"
 	svc "{{.ImportPath -}} /generated"
 )
 
-// WrapEndpoint will be called individually for all endpoints defined in
-// the service. Implement this with the middlewares you want applied to
-// every endpoint.
-func WrapEndpoint(in endpoint.Endpoint) endpoint.Endpoint {
-	return in
-}
-
 // WrapEndpoints takes the service's entire collection of endpoints. This
 // function can be used to apply middlewares selectively to some endpoints,
 // but not others, like protecting some endpoints with authentication.
 func WrapEndpoints(in svc.Endpoints) svc.Endpoints {
+
+	// Apply a middleware selectively
+	// in.Echo = endpoint.Middleware(in.Echo)
```

Comment:
This is confusing, because as written, this is a type conversion of `in.Echo` to type `endpoint.Middleware`. The intent, though, is is to show a middleware instance--a function--being invoked on `in.Echo`, an `endpoint.Endpoint`. Make up a dummy middleware like `accessLoggingMiddleware`.

Additionally, the `in.Echo` example here is only going to make sense to someone who is familiar to the example, and seems out of place in a completely different service. Better, perhaps, to use a template like the following:
```
{{- $ep := (printf "in.%s" (index .Service.Endpoints 0)) }}
{{$ep}} = accessLoggingMiddleware({{$ep}})
```
This way, an actual endpoint from the generated service is shown in the example.